emop-core/src/main/java/edu/cornell/emop/util/Util.java

```diff
@@ -102,4 +103,66 @@ public static Set<String> retrieveProjectPackageNames(File classesDir) {
         Set<String> fullSet = classFilesWalk(classesDir, classesDir.getAbsolutePath());
         return fullSet;
     }
+
+    /**
+     * Analyzes a violations file and returns a set of violations.
+     *
+     * @param violationsPath The file where violations are located
+     * @return A set of violations, each violation is a list containing the specification, a class, and a line number
+     */
+    public static Set<List<String>> parseViolations(String violationsPath) {
+        try {
+            return Files.readAllLines(new File(violationsPath).toPath())
+                    .stream()
+                    .map(Util::parseViolation)
+                    .collect(Collectors.toSet());
+        } catch (IOException exception) {
+            return new HashSet<>();
+        }
+    }
+
+    /**
+     * Analyzes a violations file and returns a list of violations.
```

Comment:
I think you might want to reword this to say: Analyzes a violations file and returns the set of specifications that were violated.